### PR TITLE
Connor #28

### LINKED ISSUE #28: Add user feedback to approve or disapprove the result

### PR DESCRIPTION
**Pull request by connorjoleary**
#28

### CODE REVIEW

File: extension/js/tree.js [hunk 1]
```diff
@@ -2,6 +2,11 @@
  *	DeepCite Source Tree page 
 */
 
+const url = "https://us-central1-deepcite-306405.cloudfunctions.net/deepcite";
```
Comment (deadmau6) [inline extension/js/tree.js]: Eventually, do you think these could be environment variables that are loaded in when building the code?

File: extension/js/tree.js [hunk 2]
```diff
@@ -88,22 +98,77 @@ function generateTestData() {
...
+			stage: stageValue}
 	console.log("upvote clicked on citeID " + citeID);
+
+	// disable button
+	element.classList.add("btn-disabled");
+	element.innerText = "Sending...";
+
+	// Submit feedback
+	// Code used to run locally
+	data['test']=true
```
Comment (deadmau6) [inline extension/js/tree.js]: Noice Comments 👍

File: extension/js/tree.js [hunk 3]
```diff
@@ -88,22 +98,77 @@ function generateTestData() {
...
+}
+
+async function upvoteButtonClicked(event) {
+	var ipValue = await grab_ip();
+
+	var element = event.srcElement
 	var citeID = findClosestCiteID(element);
+
+	data = {type: "source",
+			ip: ipValue,
```
Comment (deadmau6) [inline extension/js/tree.js]: Oof spacing, we should probably set up [prettierjs](https://prettier.io/). which I can do and also you can add it to git hooks so you don't have to remember to use it all the time!

File: backend/lambda/main.py [hunk 4]
```diff
@@ -42,12 +42,31 @@ def lambda_handler(event):
...
+        source_id = event.get('sourceId')
+        base_id = event.get('baseId')
+        
+        results = f"The source: {source_id} for base id: {base_id} has been noted"
+        error = None
+        try:
+            database_calls.record_source(base_id, source_id, user_id, stage, versions)
+        except Exception as e:
+            print("Unable to store source")
+            traceback.print_tb(e.__traceback__)
```
Comment (deadmau6) [inline backend/lambda/main.py:64]: Whoa! Noice, never seen `print_tb` before.

**Review by deadmau6 — APPROVED** [review]

File: extension/js/tree.js (hunk 1 above)
Comment (connorjoleary) [reply extension/js/tree.js]: https://stackoverflow.com/questions/196498/how-do-i-load-the-contents-of-a-text-file-into-a-javascript-variable

Comment (connorjoleary) [reply extension/js/tree.js]: A good idea. I'll look into doing this

**Pull request merged**